### ISSUE TITLE
Improved responsive display required for 'my-records' report

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/ABLE#121]
https://butterfly-monitoring.net/elastic/my-records

Improve display when browsing from mobile devices, e.g. make the table content vertical and require user to scroll up/down to view

**Comment by johnvanbreda** [comment]
Is the solution here to improve the implementation of Footable? This means that when the page is too narrow the + button appears so you can see the hidden columns. I can improve this so it automatically prioritises the columns from left to right, hiding those on the right first.

Or, another improvement might be an option to display the extra cells under the row by default rather than having to expand each row individually. There is a tip on doing this at https://stackoverflow.com/questions/23232490/footable-expand-all-rows (for my own info).

**Comment by DavidRoy** [comment]
Thanks.  Improved use of Footable seems a good option.
@larspett may have a view

**Comment by johnvanbreda** [comment]
I've experimented some more with this and I think the Footable option is OK. However, on the current site there is a module called Simple Responsive Table which tries to implement responsive tables across the Drupal site. This module seems to clash with both Footable and the Bootstrap tables CSS that are on the site. The module also doesn't seem to work with AJAX populated tables since it attaches to the table cells on page load. 

Is this module really needed?

**Comment by DavidRoy** [comment]
@BirenRathod @Gary-van-Breda to advise on need for 'Simple Responsive Table'

Will need to make changes on test before deploying to live

**Comment by Gary-van-Breda** [comment]
From my POV, I wasn't aware of it either.

**Comment by johnvanbreda** [comment]
@BirenRathod what do you mean by "works nicely with footer"?

**Comment by BirenRathod** [comment]
@johnvanbreda 
At present, I'm only aware that footer is using table for displaying logo. So this module helps to display table in all devices.

**Comment by BirenRathod** [comment]
@johnvanbreda 
In regards to Bootstrap table & Footable modules, These modules are useful only with views, so for using table in text editor we need an alternative module where 'Simple responsive modules' comes in the picture. Having said that I got another module called 'Reponsive tables filter' which I can try using   to solve this problem.

**Comment by BirenRathod** [comment]
@johnvanbreda 
I tried with everything but it only works best with table. Anyway I just did test with responsive tables filter module and it just provide the same functionality except it is not same as user friendly as other. 

Do you want me to install Footable module?

**Comment by BirenRathod** [comment]
@johnvanbreda 
I can see it works but is this solution for just eBMS, if it is then it will not resolve my problem. Because I need a solution which works all the Drupal 8 sites with minimum efforts.

**Comment by johnvanbreda** [comment]
Simple Responsive Table doesn't work alongside Indicia tables. That's because Indicia tables have built in Footable functionality to make them responsive. Simple Responsive Table applies it's functionality to all tables indiscriminately and tries to change the table layout in a way that breaks when the table has Footable applied. It also seems to clash with the CSS applied by Bootstrap's .table class. So, more or less, it works for plain tables but nothing else.

Do you need this on other Drupal 8 sites for page layout which uses tables, or for actual tables (other than Indicia ones which are already responsive)? If the former then it would be better to move to using the Bootstrap grid for layout rather than tables.

**Comment by BirenRathod** [comment]
@johnvanbreda 
As I mentioned before, I found replacement module of 'Simple Responsive table', So I can disable "simple responsive module". 
Also,t to use your approach of using Bootstrap grid, I have to use those CCS styles on all every Drupal   8 website which is bit fiddly. So I need a bit simple solution and 'Responsive tables filter' module fits for that purpose.

**Comment by BirenRathod** [comment]
@johnvanbreda 
Sorry I haven't got response from you to this about easy way to implement bootstrap grid without involvement of code in by user.

**Comment by johnvanbreda** [comment]
@BirenRathod sorry, I don't really understand the question. The ABLE website theme is a Bootstrap subtheme, therefore it makes sense to use the Bootstrap grid HTML style in order to define all layouts where possible. There is a little bit of inline CSS in the example I gave, but there is also a little bit of inline CSS in the table based example set up on the live site. So this is no different surely? I think it is impossible to expect to have no CSS in themes on each site, so adding a little bit of CSS will make no real difference. 

Basically, in the current approach, you are using a table which is not semantically correct for layout, it should only be for tabular data. Then you are applying a bit of CSS to centre the content plus an extra module to make it responsive as tables are not naturally responsive. My proposal is to achieve the same layout using divs and the Bootstrap grid classes, plus a bit of CSS to centre the content and then there is no need for the extra module.

**Comment by BirenRathod** [comment]
@johnvanbreda 
I have no problem of using CSS and of course we should use Bootstrap grid HTML style. You use Bootstrap grid HTML style & CSS in the text editor which you and I know how to use it but most users don't know. so they just need button or some sort of user interface, so they can achieve same result by them shelves. 

Also, adding a CSS is fine in the style.css file, So any Boostrap grid use with the defined class would behave same way as you done for footer. 

In addition to this, I have now uninstall the 'Simple Responsive Module'. So should be fine for all Indicia base page. If you can't see that effect, I recon please create class in CSS file which will use thorough out the site for bootstrap grid.

**Comment by johnvanbreda** [comment]
@BirenRathod but normal users won't actually create their own site footers, so I don't see the relevance? It's very unusual for a normal user to edit text and insert a table.

@DavidRoy the new responsive behaviour is now implemented. If that looks OK to you this issue can be closed.

**Comment by BirenRathod** [comment]
@johnvanbreda 
Mostly they are from BRC. They don't necessary need to create footer but they do add multiple logo in it.

**Comment by DavidRoy** [comment]
@larspett can you review this please and close if resolved for you?

**Comment by larspett** [comment]
It is considerably better now but still has some issues with my screen width etc. The + & - are placed more or less on the text next to the sign and table headers sometimes end up vertical. The top right menu is often halfway out of the screen (iPhone XR). I also had a look at the verification and that page is less responsive than the MyData.

**Comment by larspett** [comment]
![Image](https://i.imgur.com/D6pmbi5.jpg)
![Image](https://i.imgur.com/1YboW5X.jpg)
![Image](https://i.imgur.com/99O7Xkq.jpg)
![Image](https://i.imgur.com/5e4R5lk.jpg)
![Image](https://i.imgur.com/hQv5S6z.jpg)

**Comment by DavidRoy** [comment]
@larspett we will not put much effort into making the verification system responsive as this tool is complex and the expectation is that it will be used on a large screen.

I suggest we close this general issue now and raise any important specific issues as they arise.  The home page text misalignment will likely be addressed by some general re-design work next year
